resteasy-jaxrs/src/main/java/org/jboss/resteasy/core/PathParamInjector.java

```diff
@@ -109,30 +109,21 @@ else if (pathSegmentList)
             return segments[segments.length - 1];
          }
       }
-      else
+      List<String> list = request.getUri().getPathParameters(!encode).get(paramName);
+      if (list == null)
       {
-         List<String> list = request.getUri().getPathParameters(!encode).get(paramName);
-         if (list == null)
-         {
-            if (extractor.defaultValue == null) throw new InternalServerErrorException(Messages.MESSAGES.unknownPathParam(paramName, request.getUri().getPath()));
-            if (extractor.isCollectionOrArray())
-            {
-               return extractor.extractValues(null);
-            }
-            else
-            {
-               return extractor.extractValue(null);
-            }
-         }
-         if (extractor.isCollectionOrArray())
-         {
-            return extractor.extractValues(list);
-         }
-         else
-         {
-            return extractor.extractValue(list.get(list.size() - 1));
-         }
-      }
+	    if (extractor.defaultValue == null) throw new InternalServerErrorException(Messages.MESSAGES.unknownPathParam(paramName, request.getUri().getPath()));
+	    if (extractor.isInjectableAsCollectionOrArray())
+	    {
+	       return extractor.extractValues(null);
+	    }
+		return extractor.extractValue(null);
+	 }
+	 if (extractor.isInjectableAsCollectionOrArray())
+	 {
+	    return extractor.extractValues(list);
+	 }
+	 return extractor.extractValue(list.get(list.size() - 1));
    }
 
```

Comment:
re: the second half of the method"

For example, consider changing

>   if (x == y)
>   {
>      return z;
>   }
>   else
>   {
>      z = 17;
>   }

to

>   if (x == y)
>   {
>      return z;
>   }
>   z = 17;

They're semantically equivalent, right? But suppose someone comes along and changes the second version to


>   if (x == y)
>   {
>      z = 19;
>   }
>   z = 17;

Now, control falls through to z = 17;, even if x == y. That wouldn't have happened before.

Not likely, but still ...